src/offline/cable_output.F90

```diff
@@ -1633,1266 +1651,526 @@ SUBROUTINE write_output(dels, ktau, met, canopy, casaflux, casapool, casamet, ss
+
+    ! cplant
+    CALL generate_out_write_acc(output%casa, ovid%PlantCarbLeaf, 'PlantCarbLeaf', out%PlantCarbLeaf, REAL(casapool%cplant(:, 1)/1000.0, 4) &
+                                , ranges%TotLittCarb, patchout%PlantCarbLeaf, met)
+    CALL generate_out_write_acc(output%casa, ovid%PlantCarbWood, 'PlantCarbWood', out%PlantCarbWood, REAL(casapool%cplant(:, 2)/1000.0, 4) &
+                                , ranges%TotLittCarb, patchout%PlantCarbWood, met)
+    CALL generate_out_write_acc(output%casa, ovid%PlantCarbFineRoot, 'PlantCarbFineRoot', out%PlantCarbFineRoot, REAL(casapool%cplant(:, 3)/1000.0, 4) &
+                                , ranges%TotLittCarb, patchout%PlantCarbFineRoot, met)
+
+    CALL generate_out_write_acc(output%casa, ovid%TotLivBiomass, 'TotLivBiomass', out%TotLivBiomass, REAL((SUM(casapool%cplant, 2)) &
+                                                                                                          /1000.0, 4), ranges%TotLivBiomass, patchout%TotLivBiomass, met)
 
-       ! Add current timestep's value to total of temporary output variable:
-       out%PlantTurnoverWoodDist = out%PlantTurnoverWoodDist + &
-            REAL((casaflux%Cplant_turnover_disturbance)/86400.0 &
-            / 1.201E-5, 4)
-       IF(writenow) THEN
-          ! Divide accumulated variable by number of accumulated time steps:
-          out%PlantTurnoverWoodDist = out%PlantTurnoverWoodDist / REAL(output%interval, 4)
-          ! Write value to file:
-          CALL write_ovar(out_timestep, ncid_out, ovid%PlantTurnoverWoodDist, 'PlantTurnoverWoodDist', &
-               out%PlantTurnoverWoodDist,    &
-               ranges%NEE, patchout%PlantTurnoverWoodDist, 'default', met)
-          ! Reset temporary output variable:
-          out%PlantTurnoverWoodDist = 0.0
-       END IF
+    IF (cable_user%sync_nc_file) &
+      ok = NF90_SYNC(ncid_out)
 
-       ! Add current timestep's value to total of temporary output variable:
-       out%PlantTurnoverWoodCrowding = out%PlantTurnoverWoodCrowding + &
-            REAL((casaflux%Cplant_turnover_crowding)/86400.0 &
-            / 1.201E-5, 4)
-       IF(writenow) THEN
-          ! Divide accumulated variable by number of accumulated time steps:
-          out%PlantTurnoverWoodCrowding = out%PlantTurnoverWoodCrowding / REAL(output%interval, 4)
-          ! Write value to file:
-          CALL write_ovar(out_timestep, ncid_out, ovid%PlantTurnoverWoodCrowding, 'PlantTurnoverWoodCrowding', &
-               out%PlantTurnoverWoodCrowding,    &
-               ranges%NEE, patchout%PlantTurnoverWoodCrowding, 'default', met)
-          ! Reset temporary output variable:
-          out%PlantTurnoverWoodCrowding = 0.0
-       END IF
+  END SUBROUTINE write_output
 
+  SUBROUTINE acc_out_var_d1(output_var, out_var, acc_val)
+     LOGICAL, INTENT(IN) :: output_var ! Whether to write the current variable
+     REAL(4), DIMENSION(:) :: out_var
+     REAL(4), DIMENSION(:) :: acc_val
```

Comment:
Could you update all these new routines to remove the DIMENSION keyword as below:

```suggestion
     REAL(4) :: out_var(:)
     REAL(4) :: acc_val(:)
```

Reply:
Did for new routines as well for some old routines, but needs to be changed in many places. We can look into this in the future.

Reply:
I don't mind yet for the old stuff, just trying to make sure new code we introduce follows the standards we will want to enforce.